### ISSUE TITLE
Shift-m shortcut not working for folders in version 6.3 of QF and 115 of TB.

### ISSUE THREAD

**Issue by swamba** [RealRaven2000/QuickFolders#436]
/After upgrading to Thunderbird 115 (Fedora 38) and Quickfolders 6.3 then using the Shift-M shortcut to move mails and folders only works for mails and not folders.

If I have the Error Console open while pressing Shift-M and having selected a folder in the folder pane, then the following error appears:

> 
>Uncaught ReferenceError: gFolderTreeView is not defined                                              quickfolders-interface.js:1870:31
>    windowKeyPress chrome://quickfolders/content/quickfolders-interface.js:1870
>    keyListen chrome://quickfolders/content/quickfolders.js:719
>[Learn More]
>    windowKeyPress chrome://quickfolders/content/quickfolders-interface.js:1870
>    keyListen chrome://quickfolders/content/quickfolders.js:719

**Comment by RealRaven2000** [comment]
Looks like the global variable `gFolderTreeView`  has been fully deprecated. Please try the fixed version below:

[QuickFolders-mx-6.4pre38.zip](https://github.com/RealRaven2000/QuickFolders/files/13477115/QuickFolders-mx-6.4pre38.zip)

PS: should there be an option to remove the prompt "really move folder xx"? This was to avoid accidentally moving folders when the user really wanted to move emails. It made more sense when we were still able to select multiple folders in one operation.

-----

To test the version above, download the zip file, drag the file into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by RealRaven2000** [comment]
Note: I must remove all uses of `gFolderTreeView` . 

Functions that still use this:

```js
QuickFolders.FolderTree.refreshTree()
QuickFolders.quickMove.execute()       // this used to call it in the past to reselect the folder, when in single message tab mode
QuickFolders.renameFolder()   
  // monkey patch of gFolderTreeController.renameFolder.  was replaced with gFolderTreeController
  // gFolderTreeController seems also to be deprecated!
QuickFolders.Model.addFolder() // used to restore iconURI from tree when creating a new tab

```

After removing most of this code, the only thing left is `QuickFolders.FolderTree.refreshTree()` - but that one is only called from the debug menu item "Force Tree Refresh", so it's a lower priority right now:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/3fbfd2fa-37f3-430d-8da2-c2d0b853747c)

**Comment by fredstluka** [comment]
Axel,

If you add an option to remove the prompt "really move folder xx", please leave the default that the prompt DOES occur.  I use Shift-M dozens of times per day to move mail messages, but when I accidentally hit it with a folder selected, I really appreciate being prompted because I almost never want to move a folder.

Thanks!
--Fred

**Comment by RealRaven2000** [comment]
> If you add an option to remove the prompt "really move folder xx", please leave the default that the prompt DOES occur. I use Shift-M dozens of times per day to move mail messages, but when I accidentally hit it with a folder selected, I really appreciate being prompted because I almost never want to move a folder.

Agreed. I would even go further to reset it to the default behavior (ask) when Thunderbird is restarted. This is really just when you intend to do a big reorganization of many folders ( multiple selection deprecated in 115 due to [Bug 1817605](https://bugzilla.mozilla.org/show_bug.cgi?id=1817605) )

**Comment by RealRaven2000** [comment]
Fixed in 6.4 released on 04/12/2023
